prow/entrypoint/run.go

```diff
@@ -196,12 +197,12 @@ func (o Options) ExecuteProcess() (int, error) {
 	return returnCode, commandErr
 }
 
-func (o *Options) mark(exitCode int) error {
+func (o *Options) Mark(exitCode int) error {
 	content := []byte(strconv.Itoa(exitCode))
 
-	// create temp file in the same directory as the desired marker file
+	// create temp file outside the directory of the desired marker file
 	dir := filepath.Dir(o.MarkerFile)
-	tempFile, err := ioutil.TempFile(dir, "temp-marker")
+	tempFile, err := ioutil.TempFile("", "temp-marker")
 	if err != nil {
 		return fmt.Errorf("could not create temp marker file in %s: %v", dir, err)
```

Comment:
I think the use of `dir` here is misleading/wrong. And it looks like the `dir` local var is now unused. (I would expect the Go compiler to warn/error about the latter.)

Reply:
Your right, I should remove `dir`. Go compiler didn't complain because it was accidentally left in the error msg.